src/main/java/io/codelex/flightplanner/flights/FlightsDatabaseService.java

```diff
@@ -0,0 +1,106 @@
+package io.codelex.flightplanner.flights;
+
+import io.codelex.flightplanner.flights.domain.Airport;
+import io.codelex.flightplanner.flights.domain.Flight;
+import io.codelex.flightplanner.flights.exception.DepartureAndArrivalAirportAreTheSameException;
+import io.codelex.flightplanner.flights.exception.FlightAlreadyExistException;
+import io.codelex.flightplanner.flights.exception.NothingFoundException;
+import io.codelex.flightplanner.flights.request.AddFlightRequest;
+import io.codelex.flightplanner.flights.request.SearchFlightRequest;
+import io.codelex.flightplanner.flights.response.SearchFlightsResponse;
+import jakarta.transaction.Transactional;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+
+public class FlightsDatabaseService implements FlightsService {
+
+    private FlightsRepository flightsRepository;
+
+    private AirportRepository airportRepository;
+
+    public FlightsDatabaseService(FlightsRepository flightsRepository, AirportRepository airportRepository) {
+        this.flightsRepository = flightsRepository;
+        this.airportRepository = airportRepository;
+    }
+
+    @Override
+    @Transactional
+    public Flight addFlight(AddFlightRequest addFlightRequest) throws DepartureAndArrivalAirportAreTheSameException, IllegalArgumentException, FlightAlreadyExistException, NullPointerException {
+        if (addFlightRequest.isAirportFromAndToEqual()) {
+            throw new DepartureAndArrivalAirportAreTheSameException("Departure and arrival airport are  the same!");
+        }
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+        if (!addFlightRequest.isCorrectArrivalTime()) {
+            throw new IllegalArgumentException();
+        }
+        if (!this.airportRepository.existsByAirportAndCityAndCountry(addFlightRequest.getFrom().getAirport(),
+                addFlightRequest.getFrom().getCity(),
+                addFlightRequest.getFrom().getCountry())) {
+            Airport from = new Airport();
+            from.setAirport(addFlightRequest.getFrom().getAirport());
+            from.setCity(addFlightRequest.getFrom().getCity());
+            from.setCountry(addFlightRequest.getFrom().getCountry());
+            this.airportRepository.save(from);
```

Comment:
Man šķiet te jaunu objektu nav nepieciešams veidot - jebšu varēji vienkārši:
 this.airportRepository.save(addFlightRequest.getFrom());
 
 Tas pats ar toAirport.